File: csharp/bubble_sort.cs
 /**
 *  Problema: Bubble Sort
 * 
 *  n = number of elements array
 *
 *  tempo: O(nˆ2)
 *  espaço: O(1)
 */
public class bubble_sort {

    public static int[] BubbleSort(int[] nums) {
        int N = nums.Length;
        for (int i = 0; i < N; i++) {
            bool swapped = false;
            for (int j = 0; j < N - 1 - i; j++) {
                if (nums[j] > nums[j + 1]) {
                    Swap(nums, j, j + 1);
                    swapped = true;
                }
            }
            // Caso nenhum elemento tenha sido trocado na passagem atual, então o array já está ordenado
            if (!swapped) {
                break;
            }
        }
        return nums;
    }

    private static void Swap(int[] nums, int a, int b) {
        int aux = nums[a];
        nums[a] = nums[b];
        nums[b] = aux;
    }


    static void Main() {
        int[] nums = {-5, 0, 2, 8, 13, 16, 19, 23, 29, 34, 38};
        nums = BubbleSort(nums);
        foreach (int num in BubbleSort(nums)) Console.Write(num + " ");
        Console.WriteLine();

        int[] nums1 = {-5, 0, 2, 8, 13, 16, 19, 23, 29, 34, 38};
        nums1 = BubbleSort(nums1);
        foreach (int num in BubbleSort(nums1)) Console.Write(num1 + " ");
    }


}

File: ordenacao-busca/javascript/bubble_sort.js
 /**
 *  Problema: Bubble Sort
 * 
 *  n = number of elements array
 *
 *  tempo: O(nˆ2)
 *  espaço: O(1)
 */

const bubble_sort = (nums) => {
    const N = nums.length;
    for (let i = 0; i < N; i++) {
        let swapped = false;
        for (let j = 0; j < N - 1 - i; j++) {
            if (nums[j] > nums[j + 1]) {
                swap(nums, j, j + 1);
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
    return nums;
};

const swap = (nums, a, b) => {
    const aux = nums[a];
    nums[a] = nums[b];
    nums[b] = aux;
};

console.log(bubble_sort([20, 9, 86, -2, 16]));
console.log(bubble_sort([5, 4, 3, 2, 1]));
console.log(bubble_sort([-5, 0, 2, 8, 13, 16, 19, 23, 29, 34, 38]));